File: personal-assistant-web/src/api/devlog.ts
import http from './http';

export interface DevLogSummary {
  id: number; title: string; projectName: string; branchName?: string; commitHash?: string;
  tags?: string; source: string; occurredAt: string; createdAt: string;
}

export interface DevLog extends DevLogSummary {
  repository?: string; taskGoal: string; coreChanges: string; technicalDecisions?: string;
  problemSolution?: string; verificationResult?: string; markdownContent: string;
}

export interface PersonalAccessToken {
  id: number; name: string; tokenPrefix: string; scope: string; expiresAt?: string;
  lastUsedAt?: string; revokedAt?: string; createdAt: string;
}

export interface CreatedPersonalAccessToken {
  id: number; name: string; token: string; scope: string; expiresAt?: string; createdAt: string;
}

export const listDevLogs = (params: { keyword?: string; projectName?: string } = {}) =>
  http.get<unknown, { data: DevLogSummary[] }>('/devlogs', { params });
export const getDevLog = (id: number) => http.get<unknown, { data: DevLog }>(`/devlogs/${id}`);
export const listPersonalAccessTokens = () =>
  http.get<unknown, { data: PersonalAccessToken[] }>('/personal-access-tokens');
export const createPersonalAccessToken = (data: { name: string; expiresAt?: string }) =>
  http.post<unknown, { data: CreatedPersonalAccessToken }>('/personal-access-tokens', data);
export const revokePersonalAccessToken = (id: number) =>
  http.patch(`/personal-access-tokens/${id}/revoke`);

File: personal-assistant-server/src/main/java/com/personal/assistant/module/devlog/entity/DevLog.java
package com.personal.assistant.module.devlog.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("dev_log")
public class DevLog {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String fingerprint;
    private String title;
    private String projectName;
    private String repository;
    private String branchName;
    private String commitHash;
    private String taskGoal;
    private String coreChanges;
    private String technicalDecisions;
    private String problemSolution;
    private String verificationResult;
    private String tags;
    private String source;
    private LocalDateTime occurredAt;
    private String markdownContent;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
